src/month_calendar/index.es

```diff
@@ -0,0 +1,185 @@
+  });
+
+  const nextClasses = clsx('c-datepicker-controls-next', {
+    'is-disabled': !onNext,
+  });
+
+  return (
+    <div className="c-datepicker-controls">
+      <i onClick={handlePrevious} className={previousClasses}>
+        ←
+      </i>
+      <span className="c-datepicker-controls-label">{label}</span>
+      <i onClick={handleNext} className={nextClasses}>
+        →
+      </i>
+    </div>
+  );
+};
+
+const DAYS_COUNT = 7;
+
+const renderLabelsRow = (locale) => {
+  const { firstDay, modules: { calendar: { weekdaysShort } } } = locale;
+  const labels = arrayOf(DAYS_COUNT).map((i) => {
+    const key = i + firstDay > 6 ? 0 : i + firstDay;
+    const className = clsx('c-datepicker-label', {
+      'is-weekend': key === 6 || key === 0,
+    });
+    return <th key={key} className={className}>{weekdaysShort[key]}</th>;
+  });
+  return <tr>{labels}</tr>;
+};
+
+const renderMonth = ({ month, minDate, maxDate, selected, onCellClick }) => {
+  const { days, offset } = getMonthDetails(month);
+
+  const weeks = Math.ceil((days + offset) / DAYS_COUNT);
+  const today = getToday();
+
+  const renderWeek = (week) => {
+    const shift = week * DAYS_COUNT;
+    const isFirstWeek = week === 0;
+
+    const items = arrayOf(DAYS_COUNT).map((day) => {
+      const date = shift + day + 1 - offset;
+
+      const key = `${month}-${pad(date)}`;
+
+      const isBeforeMinDate = minDate ? isBefore(key, minDate) : false;
+      const isAfterMaxDate = maxDate ? isAfter(key, maxDate) : false;
+
+      const isDisabled = isBeforeMinDate || isAfterMaxDate;
+
+      const isStartFiller = isFirstWeek && day < offset;
+      const isEndFiller = date > days;
+
+      const className = clsx('c-datepicker-cell', {
+        'is-today': today === key,
+        'is-selected': key === selected,
+        'is-disabled': isDisabled,
+        'is-interactive': !isDisabled && !isStartFiller && !isEndFiller
+      });
+
+      let label;
+      if (!isStartFiller && !isEndFiller) {
+        const handleClick = onCellClick && !isDisabled ? onCellClick.bind(null, key) : null;
+        label = <span className="c-datepicker-date" onClick={handleClick}>{date}</span>;
+      }
+
+      return <td key={date} className={className}>{label}</td>;
+    });
+
+    return <tr key={week}>{items}</tr>;
+  };
+
+  return arrayOf(weeks).map(renderWeek);
+};
+
+const Calendar = ({ locale, month, minDate, maxDate, selected, onCellClick }) => {
+  const rows = renderMonth({ month, minDate, maxDate, selected, onCellClick });
+  return (
+    <table className="c-datepicker-table" cellSpacing="0">
+      <thead>{renderLabelsRow(locale)}</thead>
+      <tbody>{rows}</tbody>
+    </table>
+  );
+};
+
+const MonthCalendar = ({
+  className: passedClassName,
+  locale,
+  defaultMonth,
+  selected,
+  onChange,
+  minDate,
+  maxDate,
+  yearRange,
+}) => {
+  const [month, setMonth] = useState(defaultMonth);
+
+  useEffect(() => {
+    setMonth(defaultMonth);
+  }, [defaultMonth]);
```

Comment:
if defaultMonth changes it will change state value. Is it intended?

Reply:
Yes it is intended. I would expect the inner month to change if I change defaultMonth prop. 

I assume you would expect `month` to always override `defaultMonth` even if it changed?

With that method you have no control from outside the component to change the current month. Maybe you want to set a different month because some other data changed or something like that.

Reply:
why do we need defaultMonth at all?

Reply:
Datepicker component in nebenan-frontend doesn't have `month` property. It got there from copying `MonthCalender`. Therefore: Not needed